src/shared/condition.h

```diff
@@ -63,6 +63,14 @@ typedef struct Condition {
         LIST_FIELDS(struct Condition, conditions);
 } Condition;
 
+typedef struct SystemCondition {
+        Condition *match_host;
+        Condition *match_virt;
+        Condition *match_kernel_cmdline;
+        Condition *match_kernel_version;
+        Condition *match_arch;
+} SystemCondition;
```

Comment:
Hmm, so the condition concept is shared with PID 1 for units, but behaves quite differently for no great reason. I wonder if it it wouldn't be better to rework this so that both schemes work more alike. Specifically, instead of introducing this structure above, maybe just use the fact that Condition objects can be joined in a linked list anyway, and allow multiple specification of them, like in units?

(I mean, I don't think everything permitted in units should really be permitted in .network files too. For example, the "trigger" concept of unit conditions might not be appropriate in network files, and maybe the set of condition types to support in network files should be smaller, but still, I think the infrastructure for both should be similar...)

Reply:
I've posted PR #11931.